### ISSUE TITLE
Use for patchelf?

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Role-Dino#2]
interesting?

https://github.com/NixOS/patchelf

**Comment by kiwiroy** [comment]
My experience suggests it works well.

**Comment by plicease** [comment]
@shawnlaffan has alienized patchelf as `Alien::patchelf` https://metacpan.org/release/SLAFFAN/Alien-patchelf-0.01/view/lib/Alien/patchelf.pm

**Comment by shawnlaffan** [comment]
My immediate plan is to use it for autotool aliens for which passing LDFLAGS like ```-Wl,rpath=\$ORIGIN/../lib``` leads to shell escape headaches.  

Let me know if it would be sensible for it to live under the PerlAlien org.

**Comment by shawnlaffan** [comment]
Oh, and passing relative and absolute rpath args in the build process seems to work well with Alien::geos::af and Alien::proj.  Both use Cmake, which handles the shell escapes sanely.  

Example: https://github.com/shawnlaffan/perl-alien-proj/blob/71970eb98c73e69d08f5fef0fc8561034ad269bc/alienfile#L293-L303

**Comment by plicease** [comment]
Relative paths for linking is I think what I had in mind, makes relocatable `@INC` workable.  Haven't made any progress beyond just thinking about it mind you.

I'd be happy to host these sort of Aliens under the PerlAlien org!

**Comment by shawnlaffan** [comment]
> I'd be happy to host these sort of Aliens under the PerlAlien org!

Let me know what I need to do.

**Comment by shawnlaffan** [comment]
> Relative paths for linking is I think what I had in mind, makes relocatable @INC workable

Absolute paths can be useful, at the least on cpan-testers where some testers keep everything in their respective blib dirs and not the final install location.  The approach I use at the moment is to have the relative paths first.  If they don't exist then they are skipped anyway.

**Comment by plicease** [comment]
> Let me know what I need to do.

I think all you need to do is go to https://github.com/shawnlaffan/perl-alien-patchelf/settings and under "Danger Zone" (scary lol) there is a button that says Trasfer.  (If you do, please change the name to Alien-patchelf to match the convention of the other repos).  I think as a member of the org you should be able to do this without my intervention (and create new repos as well), but if not let me know and I will bump your access.

**Comment by shawnlaffan** [comment]
> Ah nice. So you can specify multiple paths?

Yes, it uses the same colon separated format as normal unix paths.

**Comment by shawnlaffan** [comment]
Repo transferred but I can no longer change its name.
